packages/fxa-auth-db-mysql/lib/db/schema/patch-097-098.sql

```diff
@@ -0,0 +1,148 @@
+--
+-- This migration introduces subscriptions enabled for accounts
+--
+
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('97');
+
+CREATE TABLE IF NOT EXISTS accountSubscriptions (
+  id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  uid BINARY(16) NOT NULL,
+  productName VARCHAR(191), -- TBD pending subhub docs & integration
+  subscriptionId VARCHAR(191), -- TBD pending subhub docs & integration
+  createdAt BIGINT SIGNED NOT NULL,
+  UNIQUE KEY accountSubscriptionsSubscriptionIdUnique(subscriptionId),
+  UNIQUE INDEX accountSubscriptionsUnique(uid, productName, subscriptionId)
+) ENGINE=InnoDB;
+
+CREATE PROCEDURE `createAccountSubscription_1` (
+  IN inUid BINARY(16),
+  IN inSubscriptionId VARCHAR(191),
+  IN inProductName VARCHAR(191),
+  IN inCreatedAt BIGINT SIGNED
+)
+BEGIN
+
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  SET @accountCount = 0;
+
+  -- Signal error if no user found
+  SELECT COUNT(*) INTO @accountCount FROM accounts WHERE uid = inUid;
+  IF @accountCount = 0 THEN
+    SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1643, MESSAGE_TEXT = 'Can not create subscription for unknown user.';
+  END IF;
+
+  INSERT INTO accountSubscriptions(
+    uid,
+    subscriptionId,
+    productName,
+    createdAt
+  )
+  VALUES (
+    inUid,
+    inSubscriptionId,
+    inProductName,
+    inCreatedAt
+  );
+
+  COMMIT;
+END;
+
+CREATE PROCEDURE `deleteAccountSubscription_1` (
+  IN inUid BINARY(16),
+  IN inSubscriptionId VARCHAR(191)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM accountSubscriptions
+  WHERE
+    uid = inUid
+  AND
+    subscriptionId = inSubscriptionId;
```

Comment:
Fwiw I saw that earlier comment went by about putting this inside a transaction and it surprised me, interested to hear what the reasoning is if whoever's reading. Looks like it would be fine as a lone query (to my admittedly non-expert eye).

Reply:
Looks like [that was](https://github.com/mozilla/fxa/pull/775#discussion_r273511574) @vbudhram - I just went ahead and did it since I don't quite understand why but it doesn't seem to hurt 😅